src/guc.c

```diff
@@ -546,6 +548,32 @@ _guc_init(void)
 							 NULL);
 #endif
 
+	DefineCustomStringVariable(/* name= */ MAKE_EXTOPTION("compression_segmentby_default_function"),
+							   /* short_desc= */ "Function that sets default segment_by",
+							   /* long_desc= */
+							   "Function to use for calculating default segment_by setting for "
+							   "compression",
+							   /* valueAddr= */ &ts_guc_default_segmentby_fn,
+							   /* Value= */ "_timescaledb_functions.get_segmentby_defaults",
+							   /* context= */ PGC_USERSET,
+							   /* flags= */ 0,
+							   /* check_hook= */ NULL,
+							   /* assign_hook= */ NULL,
+							   /* show_hook= */ NULL);
+
+	DefineCustomStringVariable(/* name= */ MAKE_EXTOPTION("compression_orderby_default_function"),
+							   /* short_desc= */ "Function that sets default order_by",
+							   /* long_desc= */
+							   "Function to use for calculating default order_by setting for "
+							   "compression",
+							   /* valueAddr= */ &ts_guc_default_orderby_fn,
+							   /* Value= */ "_timescaledb_functions.get_orderby_defaults",
+							   /* context= */ PGC_USERSET,
+							   /* flags= */ 0,
+							   /* check_hook= */ NULL,
```

Comment:
Do you want to add a check that the function actually exists?

Reply:
fixed